shopfloor_mobile_base/static/wms/src/components/datepicker/date_picker.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright 2022 Camptocamp SA (http://www.camptocamp.com)
+ * License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
+ */
+
+import event_hub from "../../services/event_hub.js";
+
+export var DatePicker = Vue.component("date-picker-input", {
+    data: function () {
+        return {
+            date: "",
+        };
+    },
+    watch: {
+        date: function () {
+            this.$root.trigger("date_picker_selected", this.date);
```

Comment:
Either you use `event_hub.$emit` or `this.$root.trigger`.
I don't think you need an event which has a global state here -> you don't need `$root.trigger`.